Library/ViewModels/AddNewCardViewModelTests.swift

```diff
@@ -405,9 +402,53 @@ internal final class AddNewCardViewModelTests: TestCase {
   func testReusableCardSwitchisOnByDefault_Settings() {
     self.rememberThisCardToggleViewControllerIsOn.assertDidNotEmitValue()
 
-    self.vm.inputs.configure(with: .settings)
+    self.vm.inputs.configure(with: .settings, project: nil)
     self.vm.inputs.viewDidLoad()
 
     self.rememberThisCardToggleViewControllerIsOn.assertValues([true])
   }
+
+  func testUnsupportedCardBrandsError() {
+    let project = Project.cosmicSurgery
+      |> Project.lens.availableCardTypes .~ [
+        "AMEX",
+        "MASTERCARD",
+        "VISA",
+        "UNION_PAY"
+      ]
+
+    self.vm.inputs.viewDidLoad()
+    self.vm.inputs.configure(with: .pledge, project: project)
+
+    self.creditCardValidationErrorContainerHidden
+      .assertValues([true], "Unsupported card message is hidden on viewDidLoad")
+
+    self.vm.inputs.creditCardChanged(cardDetails: ("3566", nil, nil, nil, .jcb))
+
+    self.creditCardValidationErrorContainerHidden
+      .assertValues([true, false], "Unsupported card message hides with a valid card brand")
+    self.unsupportedCardBrandErrorText
+      .assertValues(
+        ["You can’t use this credit card to back a project from Hastings, UK."],
+        "Card is unsupported"
+      )
+  }
+
+  func testUnsupportedCardBrandsError_fromSettings() {
```

Comment:
I think this test might be redundant since we already have the following tests for this intent:

`func testUnsupportedCardMessage_HiddenOnViewDidLoad()` (line 329)
`func testUnsupportedCardMessage_showsWithInvalidCardBrand_AndExistingCardNumber()` (line 336)
`func testUnsupportedCardMessage_hidesWithValidCardBrand_AndExistingCardNumber()` (line 349)
`func testUnsupportedCardMessage_hidesWithEmptyOrInvalidCardNumber()` (line 364)

The test names just need to be updated to reflect the intent that is being tested (`.settings`), and you should be able to remove this test.

Reply:
That test is testing the `cardBrandErrorText` when an invalid card is entered from the `.settings` intent. `func testUnsupportedCardMessage_HiddenOnViewDidLoad()` (line 329)